clicommand/agent_start.go

```diff
@@ -1392,8 +1396,26 @@ func handlePoolSignals(ctx context.Context, l logger.Logger, pool *agent.AgentPo
 					l.Info("Received CTRL-C, send again to forcefully kill the agent(s)")
 					pool.Stop(true)
 				} else {
-					l.Info("Forcefully stopping running jobs and stopping the agent(s)")
-					pool.Stop(false)
+					l.Info("Forcefully stopping running jobs and stopping the agent(s) in %d seconds", cancelGracePeriod)
+
+					gracefulContext, _ := job.WithGracePeriod(ctx, cancelGracePeriodDuration(cancelGracePeriod)*time.Second)
+
+					go func() {
+						l.Info("Forced agent(s) to stop")
+						pool.Stop(false) // one last chance to stop
+
+						time.Sleep(cancelGracePeriodDuration(cancelGracePeriod/2) * time.Second)
```

Comment:
`cancelGracePeriodDuration` _could_ be eliminated with the new `max` builtin:

```suggestion
						time.Sleep(max(cancelGracePeriod/2, 10) * time.Second)
```